src/gc.c

```diff
@@ -1361,44 +1378,55 @@ void gc_mark_object_list(jl_ptls_t ptls, arraylist_t *list, size_t start)
     }
 }
 
+STATIC_INLINE void gc_assert_datatype(jl_datatype_t *vt)
+{
+    if (__unlikely(jl_is_datatype(vt)))
```

Comment:
not that it affects anything on all but a tiny number of target processors with a small subset of compiler versions, but why is this `__unlikely`?